nssp/tests/test_patch.py

```diff
@@ -0,0 +1,38 @@
+import unittest
+from unittest.mock import patch as mock_patch, call
+from delphi_nssp.patch import patch
+import os
+import shutil
+
+class TestPatchModule(unittest.TestCase):
+    def test_patch(self):
```

Comment:
Can we extend the dates where you're testing right here? I recall you mentioned and also see in the patch code that there's this 
```
        current_issue_week = Week.fromdate(current_issue)
        current_issue_dir = f"""{params["patch"]["patch_dir"]}/issue_{current_issue_week}/nssp"""
 ```
 that's different from just incrementing the issue_date for the directory, the test should check if for 2-3 weeks if the date format is populated properly.
 
 I suggest looking into freezegun python package; it's already used in the google symptom tests, specifically `test_pull.py`  and ~should~ be fairly simple (famous last words)

Reply:
So I changed the directory mechanism a bit [here](https://github.com/cmu-delphi/covidcast-indicators/blob/8f0bb320c5ede9bb4bc641ea94a3a544f04f1a3d/nssp/delphi_nssp/patch.py#L73-L79) due to how our db handle weekly data coming in being quite different from daily-ingesting sources.

Basically, all weekly data coming in to our db has the `issue_date` column always marked as the first date of the epiweek that the reporting date falls into, rather than the reporting date itself. For example, I query nssp data from our api and got something like this:
```
geo_value                  signal source geo_type time_type time_value  \
0      54079  pct_ed_visits_combined   nssp   county      week 2022-09-25
**issue**  lag  missing_value  missing_stderr  missing_sample_size  value  \
0  **2024-06-30**   93              0               1                    1   3.27
```
Note the issue is `2024-06-30` despite this data coming in on July 5th. It's because `2024-06-30` is the start date of epiweek `202427`, so any data coming in that week will be marked under that issue date. 

Furthermore, the reason why we use this directory format is because of this `batch_issue_format` [implementation in acquisition](https://github.com/cmu-delphi/delphi-epidata/blob/72b4050993fa2080cae1f0570bd9f8febc013428/src/acquisition/covidcast/csv_to_database.py#L43), which demands folder format to be `issue_yyyymmdd` including for patches of weekly data.

Hence, I changed the implementation to fit with these expectations. lmk if all this makes sense.

Reply:
That makes sense; I just wanted to see if how it would work if the outage was like couple months, where you would have multiple directories for patching. Does the date really line up? Sanity checks like that

Reply:
I see. yep will add that test.